### PR TITLE
Kevin yoo/hmp 198 data transfer no public globus

### PR DESCRIPTION
**Pull request by kevinyooky**
Hello team, 

I am waiting on some items before this is finalized, but it would be helpful if I could get some feedback so I can apply them to the other authentication/access-level scenarios ASAP. 

Thanks.

### CODE REVIEW

File: context/app/static/js/pages/Dataset/Dataset.jsx [hunk 1]
```diff
@@ -134,7 +138,11 @@ function DatasetDetail({ assayMetadata, vitData, hasNotebook, visLiftedUUID }) {
     visualization: Boolean(vitData),
     protocols: Boolean(protocol_url),
     metadata: Boolean(Object.keys(combinedMetadata).length),
-    files: true,
+    files: !isAuthenticated && accessType === 'Public',
```
Comment (john-conroy) [inline context/app/static/js/pages/Dataset/Dataset.jsx:141]: I would show files if `metadata.files` exists or has a length. I forget if it's an array.

**Review by john-conroy — CHANGES_REQUESTED** [review]
I would like the directory structure to reflect other components. Let's spend some time together and walk through this.

**Review by NickAkhmetov — CHANGES_REQUESTED** [review]
Nice work! 

It looks like there were some duplicates of files that got somehow swept up/added to the commit, could you clear those out?

File: context/app/static/js/components/detailPage/BulkDataTransfer/BulkDataTransfer.jsx [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+import { useAppContext } from 'js/components/Contexts';
+import SectionHeader from 'js/shared-styles/sections/SectionHeader';
+import { DetailPageSection } from 'js/components/detailPage/style';
+import NotLoggedIn from './NotLoggedIn';
+import LoggedIn from './LoggedIn';
+
+function BulkDataTransfer() {
+  const { isAuthenticated } = useAppContext();
+  return (
+    <DetailPageSection id="bulkDataTransfer">
```
Comment (john-conroy) [inline context/app/static/js/components/detailPage/BulkDataTransfer/BulkDataTransfer.jsx:11]:
```suggestion
    <DetailPageSection id="bulk-data-transfer">
```
Use kebab casing for ids.

The id needs to reflect what's used for the table of contents on the dataset page.

**Comment by kevinyooky** [comment]
I merged main into this branch but there is still some issue with Git that is saying conflicts need to be resolved. No conflicts I can see on my end in my terminal and vscode gui. 

In any case, I addressed all the changes we discussed in our call!

**Comment by NickAkhmetov** [comment]
The conflict is the `settings.json` file still being present/tracked by git in your branch, but deleted from the repository upstream (I moved our shared settings to `default.settings.json` while configuring ESLint to allow us to choose whether we want autofix on save behavior and allow us to add other settings on an individual basis).

If you run these commands in terminal:
```
git checkout main
git pull
git checkout kevin-yoo/hmp-198-data-transfer-no-public-globus
git merge main
```

VS Code should indicate the conflict is present (hit the little refresh button above the commit message input if it doesn't show up right away):

image

Then press `+` on the `settings.json` file and hit "Delete file":

image

After that, you can commit the changes and push to resolve the merge.

File: context/app/static/js/components/detailPage/BulkDataTransfer/NotLoggedIn/ProtectedData/Links.jsx [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+import { SecondaryBackgroundTooltip } from 'js/shared-styles/tooltips';
+import { InfoIcon } from 'js/shared-styles/icons';
+import { LinkContainer, StyledLink, LinkTitle } from './style';
+
+function GlobusAccess() {
+  const {
+    entity: { dbgap_study_url, dbgap_sra_experiment_url },
+  } = useFlaskDataContext();
+
+  const LinksContentArray = [
```
Comment (NickAkhmetov) [inline context/app/static/js/components/detailPage/BulkDataTransfer/NotLoggedIn/ProtectedData/Links.jsx:16]:
```suggestion
  const linksContentArray = [
```

Capital-case (aka PascalCase) is reserved for components.

**Review by NickAkhmetov — COMMENTED** [review]
Nice job, thank you for applying our feedback!

I wrote up some next steps to template-ize the content and make further development easier, and spotted some other minor concerns. Please ping me if you have any questions!

**Review by NickAkhmetov — CHANGES_REQUESTED** [review]
We set up a template yesterday which was meant to be reused by the DbGaP/GlobusAccess components; I'm not sure I understand why the template was torn back down and copied into multiple files.

The idea behind using a template component is to handle all of these different cases by passing it different sets of props; the styling and structure are thus defined in a single place, which those props just plug into. 

If the template is copied into all these separate files, then each file will need to have updates applied to it should these components ever change, instead of just updating the single template component. For example - say that some new styling needs to be added to all of these rows. With a template, we'd be able to apply that styling change in a single place and have it be reflected in each component which uses this template, whereas with the current approach, we would need to manually update each spot they're used in.

In short: The `DbGaP` and `GlobusAccess` components should determine which title/url/tooltip/addons to provide to the template component based on the current user's authentication status/the dataset's information, then pass these props down to a shared template component.

File: CHANGELOG-hmp-198-data-transfer-no-public-globus.md [hunk 4]
```diff
@@ -0,0 +1,2 @@
+- Added links to access dataset information in dbGaP.
+- Built out the UI for Protected Data when user is not logged in.
```
Comment (lchoy) [inline CHANGELOG-hmp-198-data-transfer-no-public-globus.md:2]: The CHANGELOG-* content should read closer to Release Notes. The audience is someone who needs to know what has happened to the software or app and why. Commit messages are for communicating with folks involved in development and design and is a better forum for text like "Built out the UI for Protected Data when user is not logged in.".

It may be helpful to review the content in the Portal UI CHANGELOG.md file which is generated from these CHANGELOG-*.md files:
- https://github.com/hubmapconsortium/portal-ui/blob/main/context/app/markdown/CHANGELOG.md
- https://github.com/hubmapconsortium/search-api/releases

Comment (lchoy) [inline CHANGELOG-hmp-198-data-transfer-no-public-globus.md:2]: The CHANGELOG-* content should read closer to Release Notes. The audience is someone who needs to know what has happened to the software or app and why. Commit messages are for communicating with folks involved in development and design and is a better forum for text like "Built out the UI for Protected Data when user is not logged in."

It may be helpful to review the content in the Portal UI CHANGELOG.md file which is generated from these CHANGELOG-*.md files:
- https://github.com/hubmapconsortium/portal-ui/blob/main/context/app/markdown/CHANGELOG.md
- https://github.com/hubmapconsortium/search-api/releases

**Review by lchoy — CHANGES_REQUESTED** [review]

**Comment by kevinyooky** [comment]
Rather than creating multiple branches, I decided to finish everything on this single branch. I apologize for the long lines of code... but I wanted to get this done quickly since we are limited on time. I built out all the pages for all 6 scenarios. 

I just need help in the conditional logic for the 4 scenarios when the user is logged in. Requesting another review so I could work on them when I start my day!

**Comment by NickAkhmetov** [comment]
While I appreciate the sense of urgency behind implementing this ASAP, please take time to rest on the weekend! Burnout is no fun.

Let's work through the rest in our pair programming session today.

File: context/app/static/js/components/detailPage/BulkDataTransfer/Links.jsx [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+function GlobusAccess() {
+  const {
+    entity: { dbgap_study_url, dbgap_sra_experiment_url },
+  } = useFlaskDataContext();
+
+  const linksContentArray = [
+    {
+      title: 'dbGaP Study',
+      url: dbgap_study_url,
+      toolTip: '',
```
Comment (john-conroy) [inline context/app/static/js/components/detailPage/BulkDataTransfer/Links.jsx:21]:
```suggestion
      tooltip: '',
```
Tooltip as a single word here and elsewhere.

File: context/app/static/js/components/detailPage/BulkDataTransfer/LoggedIn/NoProtectedAccess/NoProtectedAccess.jsx [hunk 6]
```diff
@@ -0,0 +1,23 @@
+import React from 'react';
+
+import EmailIconLink from 'js/shared-styles/Links/iconLinks/EmailIconLink';
+import { StyledContainer } from '../../style';
```
Comment (john-conroy) [inline context/app/static/js/components/detailPage/BulkDataTransfer/LoggedIn/NoProtectedAccess/NoProtectedAccess.jsx:4]: If you're traversing farther than a single level, I'd prefer absolute imports using one of the aliases we've defined. `js/components/...`

File: context/app/static/js/components/detailPage/BulkDataTransfer/utils.js [hunk 7]
```diff
@@ -0,0 +1,34 @@
+function getDUAText(mapped_data_access_level) {
```
Comment (john-conroy) [inline context/app/static/js/components/detailPage/BulkDataTransfer/utils.js:1]: Could use a switch statement.

File: context/app/static/js/shared-styles/Links/index.js [hunk 8]
```diff
@@ -10,4 +11,10 @@ const LightBlueLink = styled(LinkWithoutUnderline)`
   color: ${(props) => props.theme.palette.info.main};
 `;
 
-export { LightBlueLink };
+const StyledOpenInNewRoundedIcon = styled(OpenInNewRoundedIcon)`
```
Comment (john-conroy) [inline context/app/static/js/shared-styles/Links/index.js:14]: Is this a change necessary for all instances of `OutboundLink`?

File: context/app/static/js/components/detailPage/files/Files/Files.jsx [hunk 9]
```diff
@@ -4,13 +4,12 @@ import PropTypes from 'prop-types';
...
 import SectionHeader from 'js/shared-styles/sections/SectionHeader';
 import { DetailPageSection } from 'js/components/detailPage/style';
-import GlobusSection from '../GlobusSection';
 import FileBrowser from '../FileBrowser';
 import FileBrowserDUA from '../FileBrowserDUA';
 import FilesContext from './context';
 import { MarginBottomDiv } from './style';
 
-function Files({ files, uuid, hubmap_id, visLiftedUUID }) {
+function Files({ files }) {
```
Comment (john-conroy) [inline context/app/static/js/components/detailPage/files/Files/Files.jsx:12]: Are these props no longer needed? We're still passing these props from `Dataset` to this component. 

I believe we would still want to use `visLiftedUUID` to show information for the dataset `Support` entities. We should discuss with @tsliaw.

**Review by john-conroy — CHANGES_REQUESTED** [review]
Good job incorporating some of the feedback regarding contexts and using spacing from the theme. A few more comments for us to go over.

File: context/app/static/js/components/detailPage/files/Files/Files.jsx (hunk 9 above)
Comment (tsliaw) [reply context/app/static/js/components/detailPage/files/Files/Files.jsx:12]: I realized I forgot about the situation in which we are showing the Globus links for support dataset. Please do not remove these props until I make the mockup for how that should appear in the new "Bulk Data Transfer" section.  Other than that, I'm not sure if the props are being used for other purposes such as pulling in the related donor/sample metadata.

**Comment by kevinyooky** [comment]
I am having difficult debugging the build issue. It is saying that [data-testid="files"] element is missing, but nowhere in the codebase is this rule defined.

**Comment by NickAkhmetov** [comment]
`data-testid` is a [helper prop for Cypress testing](https://www.educative.io/answers/what-is-the-data-testid-attribute-in-testing) - it was originally on the [`Files` container](https://github.com/hubmapconsortium/portal-ui/blob/5b835ffb3f2e3b3d0c901f6fc6fe0419bf8efa42/context/app/static/js/components/detailPage/files/Files/Files.jsx#L41).

Since the corresponding component is now the `bulk-data-transfer` container, I'd suggest just adding a `data-testid="bulk-data-transfer"` there and editing the corresponding test [here](https://github.com/hubmapconsortium/portal-ui/blob/5b835ffb3f2e3b3d0c901f6fc6fe0419bf8efa42/end-to-end/cypress/e2e/portal/publication-page.cy.js#L193) to point to `bulk-data-transfer` instead of `files`.

File: context/app/static/js/components/detailPage/files/Files/Files.jsx (hunk 9 above)
Comment (NickAkhmetov) [reply context/app/static/js/components/detailPage/files/Files/Files.jsx:13]: In the meantime, we should this component should be restored to its original state pending its full replacement by the BulkDataTransfer component.

File: context/app/static/js/shared-styles/Links/index.js (hunk 8 above)
Comment (NickAkhmetov) [reply context/app/static/js/shared-styles/Links/index.js:14]: This seems to have been moved here from the `style.js` in the same directory. I think that this and the `LightBlueLink` should both be in `style.js` instead of `index.js` in order to prevent `OutboundLinkButton` from importing from the index file of its own directory (which can cause a dependency loop)

File: context/app/static/js/components/detailPage/BulkDataTransfer/GlobusAccess.jsx
Comment (NickAkhmetov) [inline context/app/static/js/components/detailPage/BulkDataTransfer/GlobusAccess.jsx:1]: This file should be safe to remove after yesterday's changes 👍

File: CHANGELOG-hmp-198-data-transfer-no-public-globus.md [hunk 10]
```diff
@@ -0,0 +1,4 @@
+- Created Login section for Consortium members for Globus Access.
+- Created links to dbGap study, SRA Experiment, and SRA Bioproject for dbGaP datasets. 
+- Created links to Globus for dbGaP datasets.
```
Comment (lchoy) [inline CHANGELOG-hmp-198-data-transfer-no-public-globus.md:3]:
```suggestion
- Add links to Globus, dbGap study, SRA Experiment, and SRA Bioproject for dbGaP datasets. 
```
Minor change to use present tense and summarize changes.

File: context/app/static/js/helpers/functions.js [hunk 11]
```diff
@@ -12,6 +12,13 @@ export function capitalizeString(s) {
   return s.charAt(0).toUpperCase() + s.slice(1);
 }
 
+export function capitalizeAndReplaceDashes(s) {
```
Comment (john-conroy) [inline context/app/static/js/helpers/functions.js:15]: You could add a few unit tests to be safe.

File: context/app/static/js/components/detailPage/BulkDataTransfer/BulkDataTransfer.jsx [hunk 12]
```diff
@@ -0,0 +1,274 @@
...
+const dbGaPTooltip =
+  'The database of Genotypes and Phenotypes archive and distribute data and results from studies that have investigated the interaction of genotype and phenotype in humans.';
+const sraExpTooltip =
+  'SRA data, available through multiple cloud providers and NCBI servers, is the largest publicly available repository of high throughput sequencing data.';
+
+const dbGaPLink = {
+  title: 'dbGaP Study',
+  tooltip: dbGaPTooltip,
+  description: 'Navigate to the "Bioproject" or "Sequencing Read Archive" links to access the datasets.',
+  outBoundLink: '',
```
Comment (john-conroy) [inline context/app/static/js/components/detailPage/BulkDataTransfer/BulkDataTransfer.jsx:26]: Outbound is a single word.

**Review by john-conroy — CHANGES_REQUESTED** [review]
Great progress! Sorry for being tedious, but a few changes here I'm glad to discuss over a call.

File: context/app/static/js/components/detailPage/BulkDataTransfer/BulkDataTransfer.jsx [hunk 13]
```diff
@@ -0,0 +1,274 @@
...
+
+  const filesContext = useMemo(() => ({ openDUA, hasAgreedToDUA }), [openDUA, hasAgreedToDUA]);
+
+  const panelsToUse = useBulkDataTransferPanels();
+
+  return (
+    <FilesContext.Provider value={filesContext}>
+      <DetailPageSection id="bulk-data-transfer">
+        <SectionHeader>Bulk Data Transfer</SectionHeader>
+        <StyledContainer>
```
[the remark this answers is not in the document]
Comment (john-conroy) [reply context/app/static/js/components/detailPage/BulkDataTransfer/BulkDataTransfer.jsx:242]: Once in a separate component, it would be nice to add some stories for the cases. We can talk about this later on.

File: context/app/static/js/components/detailPage/BulkDataTransfer/BulkDataTransfer.jsx [hunk 14]
```diff
@@ -0,0 +1,274 @@
...
+  ),
+};
+
+const useBulkDataTransferPanels = () => {
+  const { isAuthenticated, isHubmapUser } = useAppContext();
+  const {
+    entity: { mapped_data_access_level: accessType, mapped_status: status, uuid },
+  } = useFlaskDataContext();
+  const hasNoAccess = useIsProtectedFile(uuid);
+  if (isAuthenticated) {
```
[the remark this answers is not in the document]
Comment (NickAkhmetov) [reply context/app/static/js/components/detailPage/BulkDataTransfer/BulkDataTransfer.jsx:177]: Good catch!

@kevinyooky  The condition for this case is if the `accessType === 'Public'`. 

My suggested approach here would be to move the `PUBLIC_DATA` case above the `isAuthenticated` case and checking for the accessType there and making the `PROTECTED_DATA` case the default case at the end instead (i.e. if the dataset is not public, and the user is not authenticated, then the "data is protected" case is displayed.)

**Comment by kevinyooky** [comment]
**Not logged in** Public and links are available:

image

image

**Comment by kevinyooky** [comment]
**Logged in** and has access to protected data (below):

http://localhost:5001/browse/dataset/68cd0ff83bac93597b4440d3efb1c23f

image
image
image

File: context/app/static/js/pages/Publication/Publication.jsx [hunk 15]
```diff
@@ -42,7 +43,7 @@ function Publication({ publication, vignette_json }) {
   };
```
Comment (NickAkhmetov) [inline context/app/static/js/pages/Publication/Publication.jsx:43]: GitHub's not letting me select it, but the `files: true` on line 41 here which makes the `files` header/section display no matter what should be swapped for `Boolean(files?.length)` to ensure the empty header is not displayed if there are no files to display in the `FileBrowser`. This is the reason for one of the failing tests.

**Comment by tsliaw** [comment]
@kevinyooky Anywhere the`BlockIcon` is used, please use `color="warning` since these are not errors, but that we're not providing the links to the users in those scenarios.

File: context/app/static/js/components/detailPage/BulkDataTransfer/BulkDataTransferPanels.jsx [hunk 16]
```diff
@@ -0,0 +1,45 @@
...
+    return <NoAccess>{panelsToUse.error}</NoAccess>;
+  }
+
+  return (
+    <>
+      {panelsToUse.panels.length > 0 &&
+        panelsToUse.panels.map((props) => <BulkDataTransferPanel {...props} key={props.title} />)}
+      {panelsToUse.links.length > 0 && (
+        <Paper>
+          <>
```
Comment (NickAkhmetov) [inline context/app/static/js/components/detailPage/BulkDataTransfer/BulkDataTransferPanels.jsx:34]: We should be able to remove this extra fragment wrapping the `map` since the `Paper` component but it's a non-blocking issue 😄

File: context/app/static/js/components/detailPage/files/Files/Files.jsx (hunk 9 above)
Comment (NickAkhmetov) [reply context/app/static/js/components/detailPage/files/Files/Files.jsx:13]: Update - `visLiftedUUID` and the related logic is now handled in `GlobusLinkContainer` (defined in the same file as the `GlobusLink`, so this concern should be resolved 👍

File: context/app/static/js/pages/Publication/Publication.jsx [hunk 17]
```diff
@@ -38,11 +39,11 @@ function Publication({ publication, vignette_json }) {
   const shouldDisplaySection = {
     visualizations: Boolean(Object.keys(vignette_json).length),
     provenance: !associatedCollectionUUID,
-    files: true,
+    files: Boolean(files?.length),
```
Comment (NickAkhmetov) [inline context/app/static/js/pages/Publication/Publication.jsx:42]: Just for consistency with `Dataset`:

```suggestion
    files: Boolean(files?.length),
    bulkDataTransfer: true
```

File: context/app/static/js/pages/Publication/Publication.jsx [hunk 18]
```diff
@@ -61,7 +62,8 @@ function Publication({ publication, vignette_json }) {
       {shouldDisplaySection.visualizations && (
         <PublicationsVisualizationSection vignette_json={vignette_json} uuid={uuid} />
       )}
-      <Files files={files} uuid={uuid} hubmap_id={hubmap_id} />
+      {shouldDisplaySection.files && <Files files={files} uuid={uuid} hubmap_id={hubmap_id} />}
+      <BulkDataTransfer files={files} uuid={uuid} hubmap_id={hubmap_id} />
```
Comment (NickAkhmetov) [inline context/app/static/js/pages/Publication/Publication.jsx:66]: Same as above, just for consistency 👍 

```suggestion
      {shouldDisplaySection.bulkDataTransfer && <BulkDataTransfer files={files} uuid={uuid} hubmap_id={hubmap_id} />}
```

**Review by NickAkhmetov — APPROVED** [review]
Just a few minor non-blocking things, looks great overall! Thank you for your hard work and persistence with this ticket, it turned out to be quite gargantuan.

**Review by john-conroy — APPROVED** [review]
Great work!

**Review by NickAkhmetov — APPROVED** [review]
🚀

**Review by john-conroy — APPROVED** [review]
